safe/impact_function/test/test_impact_function.py

```diff
@@ -682,6 +683,19 @@ def test_scenario(self, scenario_path=None, use_debug=True):
         # counted in the JSON file.
         self.assertEqual(len(outputs) - 1, expected_outputs['count'])
         self.assertEqual(len(outputs), computed_nb_outputs)
+
+        # Test deserialization
+        if test_loader:
+            LOGGER.debug('Test deserialization. Lontong')
+            output_metadata = impact_function.impact.keywords
+            new_impact_function = ImpactFunction. \
+                load_from_output_metadata(output_metadata)
+            try:
+                self.assertEquals(impact_function, new_impact_function)
+            except AssertionError as e:
+                raise AssertionError(
```

Comment:
Exceptions are very important if we know where they come from. By doing this, we lose the orignal traceback and we create a new one. Then we can't know where it has been raised before. Look for `Another case when people often clobber the traceback is when they want to add information to it, e.g.:` in http://www.ianbicking.org/blog/2007/09/re-raising-exceptions.html

Or maybe you can do:
```
except AssertionError as e:
     LOGGER.debug("Exception found in" +  scenario_path)
     raise
```
 +                raise AssertionError(

Reply:
Hmm, my bad. I just want to add the scenario file. Ok, I will  just use your suggestion. It seems I need to read the article in clear mind. Thanks.